signalfx/rest.py

```diff
@@ -355,6 +356,40 @@ def get_chart(self, id, **kwargs):
                                         **kwargs)
         return resp
 
+    # functionality related to dashboard groups
+    def get_dashboard_group(self, id, **kwargs):
+        """"Retrieve a (v2) dashboard group by id.
+        """
+        resp = self._get_object_by_name(self._DASHBOARD_GROUP_ENDPOINT_SUFFIX,
+                                        id, **kwargs)
+        return resp
+
+    def get_dashboard_groups(self, name=None, batch_size=100, **kwargs):
+        """Retrieve all (v2) dashboard groupss matching the given name; all (v2)
+        dashboard groups otherwise.
+
+        Note that this method will loop through the paging of the results and
+        accumulate all dashboard groups that match the query.
+        """
+        dashgroups = []
+        offset = 0
+        while True:
```

Comment:
I feel like this logic is repeated enough that it is worthy of being pulled out to a "paginator" function/helper.  I wouldn't consider it a blocker for this PR to have it but definitely next time we add something to this it needs to be pulled out.

Reply:
Agreed!